### PR TITLE
[Driver][Frontend] add the symbol graph dir to the supplementary file map

### PR DESCRIPTION
**Pull request by QuietMisdreavus**
Resolves rdar://75582169

When the `-emit-symbol-graph[-dir]` flags were added to the driver, they were marked as supplementary outputs, but were not added to the supplementary output map. This PR adds the symbol graph output directory to the supplementary output file map, and updates the frontend to read the directory from the map if present.

A PR to integrate this change into swift-driver is forthcoming.

### CODE REVIEW

**Comment by QuietMisdreavus** [comment]
@swift-ci Please smoke test

File: lib/Driver/Driver.cpp [hunk 1]
```diff
@@ -2943,6 +2944,10 @@ Job *Driver::buildJobsForAction(Compilation &C, const JobAction *JA,
                          options::OPT_emit_objc_header_path))
     chooseObjectiveCHeaderOutputPath(C, OutputMap, workingDirectory,
                                      Output.get());
+  
+  if (C.getArgs().hasArg(options::OPT_emit_symbol_graph)) {
+    chooseSymbolGraphOutputPath(C, OutputMap, workingDirectory, Output.get());
+  }
```
Comment (artemcm) [inline lib/Driver/Driver.cpp:2950]: Nit: Surrounding code more-or-less conforms to the LLVM Style guide which omits braces for single-statement bodies:
[Don’t Use Braces on Simple Single-Statement Bodies of if/else/loop Statements 
](https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements)

**Review by artemcm — APPROVED** [review]
The changes look good, thank you!

**Comment by QuietMisdreavus** [comment]
The Linux CI failed on `Driver/filelists.swift`, which i did modify, but it failed in a part that i left alone (in fact, the part that i based my additions from!). While i get a build going on my local machine, i'm going to try to run it again, in case it was a fluke...

@swift-ci Please smoke test Linux platform

**Comment by QuietMisdreavus** [comment]
I have a hunch about the failing test. It seems like the output is capturing a warning of some kind on Linux that is interfering with the scan that picks up the file lists. I've tweaked the command lines for those lines to make it a bit more resilient to situations like that.

@swift-ci Please smoke test

**Comment by QuietMisdreavus** [comment]
The Windows CI failed on concurrency tests, which i believe are being worked on elsewhere. I'm going ahead and merging this.

**Pull request merged**